blueocean-personalization/src/main/js/components/DashboardCards.jsx

```diff
@@ -62,17 +85,38 @@ export class DashboardCards extends Component {
                     />
                 </div>
             );
-        });
+          });
+
+        // do we have any paused pipelines?
+        const hasPausedFavorites = pausedCards.size > 0;
+        // Only show paused pipelines when we really have some
+        const inputParent = hasPausedFavorites ? (<div className="favorites-card-stack">
+              <div className="favorites-card-stack-heading"> {t('dashboardCard.input.required')}</div>
+              <TransitionGroup transitionName="vertical-expand-collapse"
+                transitionEnterTimeout={300}
+                transitionLeaveTimeout={300}
+              >
+                {pausedCards}
+              </TransitionGroup>
+          </div>) : null;
+        // in case we do not have any paused favorites we do not want the heading
```

Comment:
You could de-dup a lot of code here by just having a single function that returned the card stack, transition group and child cards. Just run two different filters against the main collection to get the paused / unpaused data sets, then pass each to the function that returns the React elements.